modules/core/arrow-core/src/main/kotlin/arrow/core/Option.kt

```diff
@@ -199,3 +199,17 @@ fun <A> Boolean.maybe(f: () -> A): Option<A> =
 fun <A> A.some(): Option<A> = Some(this)
 
 fun <A> none(): Option<A> = None
+
+fun <T> Iterable<T>.firstOrNone(): Option<T> = this.firstOrNull().toOption()
+
+inline fun <T> Iterable<T>.firstOrNone(predicate: (T) -> Boolean): Option<T> = this.firstOrNull(predicate).toOption()
+
+fun <T> Iterable<T>.singleOrNone(): Option<T> = this.singleOrNull().toOption()
+
+inline fun <T> Iterable<T>.singleOrNone(predicate: (T) -> Boolean): Option<T> = this.singleOrNull(predicate).toOption()
+
+fun <T> Iterable<T>.lastOrNone(): Option<T> = this.lastOrNull().toOption()
+
+inline fun <T> Iterable<T>.lastOrNone(predicate: (T) -> Boolean): Option<T> = this.lastOrNull(predicate).toOption()
```

Comment:
We're slowly doing away with the inlines. They don't give much microperf, increase the bytecode size on the clients, and they allow non-local returns which are scary :D